spec/junk_drawer/notifier_spec.rb

```diff
@@ -3,12 +3,38 @@
 RSpec.describe JunkDrawer::Notifier, '#call' do
   let(:notifier) { described_class.new }
 
-  it 'calls the configured notifier with the given arguments' do
+  after { JunkDrawer::Notifier.strategy = :null }
+
+  it 'calls the configured pre-defined notifier with the given arguments' do
     JunkDrawer::Notifier.strategy = :raise
     strategy = JunkDrawer::Notifier::RaiseStrategy
-    expect do
-      notifier.('foo', 'bar', 'butts')
-    end.to invoke(:call).on(strategy).with('foo', 'bar', 'butts')
-    JunkDrawer::Notifier.strategy = :nil
+    expect { notifier.('foo', 'bar', 'butts') }
+      .to invoke(:call).on(strategy).with('foo', 'bar', 'butts')
+  end
+
+  it 'calls the configured callable notifier with the given arguments' do
+    class FakeNotifier
```

Comment:
We don't want to define classes in our specs due to the fact that these end up in the global scope. Somewhat surprisingly, when you define a constant inside a block, it is not scoped to the block. If you don't need to refer to the class name for any reason, you can instead assign it to a local:

```rb
fake_notifier = Class.new do
  def self.call(...)
  end
end
```

I don't think you actually need the class constant, but if you did you could use `stub_const` to let RSpec set it up and remove it afterwards:

```rb
stub_const('FakeNotifier', fake_notifier)
```